apps/rush-lib/src/api/CommandLineConfiguration.ts

```diff
@@ -433,38 +483,42 @@ export class CommandLineConfiguration {
         }
       }
     }
+
+    let phaseIndex: number = 0;
+    for (const phase of this.phases.values()) {
+      phase.index = phaseIndex;
+      phaseIndex++;
+    }
   }
 
-  private _checkForPhaseSelfCycles(phase: IPhase, checkedPhases: Set<string> = new Set<string>()): void {
-    const phaseSelfDependencies: string[] | undefined = phase.dependencies?.self;
-    if (phaseSelfDependencies) {
-      for (const dependencyName of phaseSelfDependencies) {
-        if (checkedPhases.has(dependencyName)) {
-          const dependencyNameForError: string =
-            typeof dependencyName === 'string' ? dependencyName : '<synthetic phase>';
-          throw new Error(
-            `In ${RushConstants.commandLineFilename}, there exists a cycle within the ` +
-              `set of ${dependencyNameForError} dependencies: ${Array.from(checkedPhases).join(', ')}`
-          );
-        } else {
-          checkedPhases.add(dependencyName);
-          const dependency: IPhase | undefined = this.phases.get(dependencyName);
-          if (!dependency) {
-            return; // Ignore, we check for this separately
-          } else {
-            if (phaseSelfDependencies.length > 1) {
-              this._checkForPhaseSelfCycles(
-                dependency,
-                // Clone the set of checked phases if there are multiple branches we need to check
-                new Set<string>(checkedPhases)
-              );
-            } else {
-              this._checkForPhaseSelfCycles(dependency, checkedPhases);
-            }
-          }
-        }
+  private _checkForPhaseSelfCycles(
+    phase: IPhase,
+    searchedPhases: Set<IPhase>,
+    safePhases: Set<IPhase>
+  ): void {
+    if (safePhases.has(phase)) {
+      return;
+    }
+
+    for (const dependency of phase.phaseDependencies.self) {
+      if (searchedPhases.has(dependency)) {
+        const dependencyNameForError: string =
+          typeof dependency.name === 'string' ? dependency.name : '<synthetic phase>';
+        throw new Error(
+          `In ${RushConstants.commandLineFilename}, there exists a cycle within the ` +
+            `set of ${dependencyNameForError} dependencies: ${Array.from(
```

Comment:
```suggestion
        throw new Error(
          `In ${RushConstants.commandLineFilename}, there exists a cycle within the ` +
            `set of ${dependency.name} dependencies: ${Array.from(
```

Reply:
`dependencyNameForError` was a remnant of an old design that I didn't fully clean up.